src/imgui/ImGuiSettings.cc

```diff
@@ -85,6 +85,81 @@ void ImGuiSettings::setStyle()
 	}
 	setColors(selectedStyle);
 }
+
+bool ImGuiSettings::shortcutAction(ShortcutIndex index)
+{
+	static constexpr auto mods = std::array{
+		ImGuiKey_LeftCtrl, ImGuiKey_LeftShift, ImGuiKey_LeftAlt, ImGuiKey_LeftSuper,
+		ImGuiKey_RightCtrl, ImGuiKey_RightShift, ImGuiKey_RightAlt, ImGuiKey_RightSuper,
+		ImGuiKey_ReservedForModCtrl, ImGuiKey_ReservedForModShift, ImGuiKey_ReservedForModAlt,
+		ImGuiKey_ReservedForModSuper, ImGuiKey_MouseLeft, ImGuiKey_MouseRight, ImGuiKey_MouseMiddle,
+		ImGuiKey_MouseX1, ImGuiKey_MouseX2, ImGuiKey_MouseWheelX, ImGuiKey_MouseWheelY,
+	};
+	ImGuiIO& io = ImGui::GetIO();	
+	ImGuiKeyChord keyChord = ImGuiKey_None;
+	for (int key = ImGuiKey_NamedKey_BEGIN; key < ImGuiKey_NamedKey_END; ++key ) {
+		if (contains(mods, key)) continue; // skip: mods can't be primary keys in a KeyChord
+		if (ImGui::IsKeyPressed((ImGuiKey) key)) {
+			keyChord = key | (io.KeyCtrl ? ImGuiMod_Ctrl : 0) | (io.KeyShift ? ImGuiMod_Shift : 0)
+				| (io.KeyAlt ? ImGuiMod_Alt : 0) | (io.KeySuper ? ImGuiMod_Super : 0);
+			break;
+		}
+	}
+	if (keyChord != ImGuiKey_None) {
+		manager.getShortcuts().setShortcut(index, keyChord);
+		return true;
+	}
+	return false;
+}
+
+void ImGuiSettings::openShortcutPopup(ShortcutIndex index)
+{
+	auto shortcutName = Shortcuts::getShortcutName(index);
+	auto &shortcut = manager.getShortcuts().getShortcut(index);
+	auto popupName = strCat("shortcutModal##", shortcutName);
+
+	if (ImGui::TableNextColumn()) {
+		ImGui::Checkbox(strCat("local##", shortcutName).c_str(), &shortcut.local);
```

Comment:
All ImGui widgets indeed need a unique hierarchical name.
* One possibility to achieve that is via appending `##<something_unique>` to each name, like you do here.
* Another possibility is to introduce a new hierarchical level, where you specify a unique name just once. This is done via `ImGui::PushID()` and `ImGui::PopID()`, or via our `im::ID()` helper function.
* Because it's a very common pattern to have to create a unique name for each iteration in a loop, I've also created a helper function `im::ID_for_range()`. That will then replace the for-loop in `paintShortcuts()` (around line 1194).